src/test/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitorTest.java

```diff
@@ -121,4 +124,42 @@ public void GIVEN_certs_added_to_monitor_WHEN_expired_THEN_regenerated() throws
         X509Certificate cert2fourth = cg2.getCertificate();
         assertThat(cert2fourth, is(cert2third));
     }
+
+    @Test
+    public void GIVEN_certs_with_max_duration_WHEN_time_greater_than_default_THEN_not_generated() throws Exception {
+        Mockito.when(mockConfigurationTopics.findOrDefault(any(), any())).thenReturn(CertificatesConfig.MAX_SERVER_CERT_EXPIRY_SECONDS);
+        CertificateStore certificateStore = new CertificateStore(tmpPath);
+        certificateStore.update(TEST_PASSPHRASE, CertificateStore.CAType.RSA_2048);
+        CertificateExpiryMonitor certExpiryMonitor = new CertificateExpiryMonitor(ses, mockConnectivityInfoProvider);
+        certExpiryMonitor.startMonitor(Duration.ofMillis(100));
+        X500Name subject = new X500Name(SUBJECT_PRINCIPAL);
+        PublicKey key1 = CertificateStore.newRSAKeyPair().getPublic();
+
+        // Add certificate to monitor
+        CertificateGenerator cg1 = new ServerCertificateGenerator(subject, key1, mockCallback, certificateStore, certificatesConfig);
+        cg1.generateCertificate(Collections::emptyList);
+        certExpiryMonitor.addToMonitor(cg1);
+        X509Certificate cert1initial = cg1.getCertificate();
+
+        //Add DEFAULT_SERVER_CERT_EXPIRY_SECONDS time. cert1 should not expire.
+        Clock mockClock = Clock.fixed(Instant.now().plus(Duration.ofSeconds(CertificatesConfig.DEFAULT_SERVER_CERT_EXPIRY_SECONDS)),
+                ZoneId.of("UTC"));
+        cg1.setClock(mockClock);
+        TimeUnit.MILLISECONDS.sleep(TEST_CERT_EXPIRY_CHECK_MILLIS);
```

Comment:
Let's remove sleeps from these tests. This doesn't actually guarantee that the executor service had a chance to run, and it also slows the tests down.

The test can force the expiration check method to run, either by making the method package private or by mocking the ses